entity-service/src/main/kotlin/com/egm/stellio/entity/repository/Neo4jSearchRepository.kt

```diff
@@ -19,22 +26,47 @@ class Neo4jSearchRepository(
     override fun getEntities(
         queryParams: QueryParams,
         sub: Option<Sub>,
+        geoQuery: GeoQuery,
         contexts: List<String>
     ): Pair<Int, List<URI>> {
-        val query = if (neo4jAuthorizationService.userIsAdmin(sub))
-            QueryUtils.prepareQueryForEntitiesWithoutAuthentication(queryParams, contexts)
-        else
-            QueryUtils.prepareQueryForEntitiesWithAuthentication(queryParams, contexts)
-
         val userAndGroupIds = neo4jAuthorizationService.getSubjectGroups(sub)
             .plus(neo4jAuthorizationService.getSubjectUri(sub))
             .map { it.toString() }
 
+        val query = if (neo4jAuthorizationService.userIsAdmin(sub))
+            QueryUtils.prepareQueryForEntitiesWithoutAuthentication(queryParams, geoQuery, contexts)
+        else
+            QueryUtils.prepareQueryForEntitiesWithAuthentication(queryParams, geoQuery, contexts)
+
         val result = neo4jClient
             .query(query)
             .bind(userAndGroupIds).to("userAndGroupIds")
             .fetch()
             .all()
-        return prepareResults(queryParams.limit, result)
+
+        if (geoQuery.geoproperty.equals(JsonLdUtils.NGSILD_LOCATION_PROPERTY) &&
+            geoQuery.georel != null &&
+            geoQuery.geometry == "Point" &&
+            geoQuery.coordinates != null
+        ) {
+            val geoResult: ArrayList<Map<String, Any>> = ArrayList()
+            val geo1 = WKTReader().read(geoJsonToWkt(geoQuery.geometry, geoQuery.coordinates.toString()))
+            val georelParams = extractGeorelParams(geoQuery.georel)
+
+            result.forEach {
+                val geo2 = WKTReader().read(it["entityLocation"] as String)
+                val distance = distance(geo1, geo2) * MULTIPLY_DISTANCE
```

Comment:
in the `geoQ`, the distance is in meters. what is the unit of the result returned by `distance(...)`?

Reply:
The result of the unit is in kilometers. That's why I did *1000

Reply:
I can't find this info in the doc of the function. where did you find it?

Reply:
I did not find the info in their doc but by observing the results of the postgis function I get the same.
I know it's not very correct but I haven't found better solutions yet. (I wanted to keep a library that we already used)

Reply:
By "postgis function", are you referring to `ST_Distance`? (if so, it returns results in meters)

Reply:
Did a quick test, you get a distance in degrees.